subprojects/file-temp/build.gradle.kts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("gradlebuild.distribution.api-java")
+}
+
+description = "Utilities for working with temporary files & directories"
+
+gradlebuildJava.usedInWorkers()
+
+dependencies {
+    implementation(project(":base-annotations"))
+    implementation(project(":base-services"))
+
+    implementation(libs.inject)
+}
```

Comment:
This code needed to make its way into a variety of places, like workers and such, and I didn't want to be adding more dependencies than were strictly necessary when doing it.

Reply:
I think `:files` is already a dependency of everything where you'd need this. What does not depend on it that still needs temporary files?

Reply:
The projects `jvm-services` & `native` don't depend upon the entirety of `files` and I didn't want to create a coupling here on more than was necessary. My goal here was to apply the single-responsibility principle so that downstream consumers are coupled to the fewest dependencies. Thus allowing Gradle to continue to be able to parallelize the compilation of these components. If I were to introduce this explicit coupling, compilation wouldn't be able to happen in parallel.

Reply:
This is very much commendable, however with our current setup adding a new JAR adds some non-trivial costs, too, both at build time and at runtime. The most significant of these is the lengthening of the Windows classpath, which causes innumerable pains. We should fix this for good at some point by combining our JARs into a single fat one or something, but we don't seem to be getting around to doing this.

In any case I don't think parallel compilation for these three classes is a major issue that would warrant a separate subproject by itself.

Reply:
I ran this by @big-guy and he gave me guidance to continue with the separate project for now. You're not wrong about the issues here with the the windows paths. We've decided we can always re-combine these if we have issues moving forward.